### ISSUE TITLE
Doesn't work for NYT GZip Middleware

### ISSUE THREAD

**Issue by BourgeoisBear** [gwatts/gin-adapter#1]
gin-adapter doesn't compress the output, although [WrapHH](https://github.com/turtlemonvh/gin-wraphh) does...

```go
import (
   "github.com/gin-gonic/gin"
   "github.com/NYTimes/gziphandler"
   "github.com/gwatts/gin-adapter"
)
oRouteHandler := gin.Default()
oRouteHandler.Use( adapter.Wrap( gziphandler.GzipHandler ) )
...
```

**Comment by CAFxX** [comment]
Yes, there is a bug in gin-adapter: in https://github.com/gwatts/gin-adapter/blob/c44433c485ad7f66183923c1906da31ac0ec385f/adapter.go#L74 it only overrides the request, but it does not override the responsewriter, effectively discarding the responsewriter passed by the wrapped middleware

**Comment by gwatts** [comment]
Should be fixed courtesy of #2 - Please add a comment if it's still troublesome
